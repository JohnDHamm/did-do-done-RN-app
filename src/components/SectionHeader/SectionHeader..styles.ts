import styled from 'styled-components/native';
import { FONTS } from '../../styles';

export const Container = styled.View<{ color: string }>`
  background-color: ${(props) => props.color};
  width: 100%;
  padding: 5px 0;
  justify-content: center;
  align-items: center;
`;

export const HeaderText = styled.Text<{ textColor: string }>`
  font-family: ${FONTS.PRIMARY};
  font-size: 24px;
  color: ${(props) => props.textColor};
`;
