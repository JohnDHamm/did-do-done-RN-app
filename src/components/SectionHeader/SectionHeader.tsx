import React from 'react';
import { Container, HeaderText } from './SectionHeader..styles';
import { COLORS } from '../../styles';

interface SectionHeaderProps {
  text: string;
  color?: string;
  textColor?: string;
}

const SectionHeader: React.FC<SectionHeaderProps> = ({
  color = COLORS.PRIMARY_GRAY,
  text,
  textColor = COLORS.PRIMARY_PURPLE,
}: SectionHeaderProps) => {
  return (
    <Container color={color}>
      <HeaderText textColor={textColor}>{text}</HeaderText>
    </Container>
  );
};

export default SectionHeader;
