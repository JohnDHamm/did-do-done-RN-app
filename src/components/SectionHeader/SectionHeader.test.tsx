import React from 'react';
import { cleanup, render } from 'react-native-testing-library';
import SectionHeader from './SectionHeader';

describe('SectionHeader', () => {
  afterEach(cleanup);

  it('renders correctly with defaults', () => {
    const { toJSON } = render(<SectionHeader text="testing" />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the text prop correctly', () => {
    const { getByText } = render(<SectionHeader text="Test heading" />);

    expect(getByText('Test heading')).not.toBeNull;
  });
});
