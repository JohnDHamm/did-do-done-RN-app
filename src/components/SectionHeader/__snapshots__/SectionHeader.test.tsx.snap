// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader renders correctly 1`] = `
<View
  color="#c3c3c3"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c3c3c3",
        "justifyContent": "center",
        "paddingBottom": 5,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 5,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#4c408f",
          "fontFamily": "Lobster-Regular",
          "fontSize": 24,
        },
      ]
    }
    textColor="#4c408f"
  >
    testing
  </Text>
</View>
`;
